locator/network_topology_strategy.cc

```diff
@@ -557,6 +557,12 @@ tablet_replica_set network_topology_strategy::drop_tablets_in_dc(schema_ptr s, c
 
 sstring network_topology_strategy::sanity_check_read_replicas(const effective_replication_map& erm,
                                                               const inet_address_vector_replica_set& read_replicas) const {
+    const auto& replication_strategy = erm.get_replication_strategy();
+    if (replication_strategy.uses_tablets()) {
+        // Does not make sense since Tablets are calculating the replication factor basing on the replica set.
```

Comment:
It doesn't make sense, or it's redundant? Even if it's no longer reproducing the bug (as it got fixed), it still acts nicely to prevent regressions I think.

Reply:
Ok. I will need to fix this check then, since it does not take the token in to account.